pkgs/applications/graphics/processing/default.nix

```diff
@@ -1,73 +1,78 @@
-    sha256 = "0cvv8jda9y8qnfcsziasyv3w7h3w22q78ihr23cm4an63ghxci58";
+    repo = "processing4";
+    rev = "processing-${version}";
+    sha256 = "sha256-U6qYHpkf7yCWnrl0GfDGgIrCN62AyNarRy/A1iUDlUQ=";
   };
 
-  patches = [
-    (fetchpatch {
-      name = "oraclejdk-8u281-compat.patch";
-      url = "https://github.com/processing/processing/commit/7e176876173c93e3a00a922e7ae37951366d1761.patch";
-      sha256 = "g+zwpoIVgw7Sp6QWW3vyPZ/fKHk+o/YCY6xnrX8IGKo=";
-    })
-  ];
+  nativeBuildInputs = [ ant unzip makeWrapper wrapGAppsHook ];
+  buildInputs = [ jdk javaPackages.jogl_2_3_2 ant rsync ffmpeg batik ];
 
-  nativeBuildInputs = [ ant rsync makeWrapper ];
-  buildInputs = [ jdk ];
+  dontWrapGApps = true;
 
   buildPhase = ''
-    # use compiled jogl to avoid patchelf'ing .so files inside jars
-    rm core/library/*.jar
-    cp ${javaPackages.jogl_2_3_2}/share/java/*.jar core/library/
-
-    # do not download a file during build
-    ${xmlstarlet}/bin/xmlstarlet ed --inplace -P -d '//get[@src="http://download.processing.org/reference.zip"]' build/build.xml
-    install -D -m0444 ${fetchurl {
-                          # Use archive.org link for reproducibility until the following issue is fixed:
-                          # https://github.com/processing/processing/issues/5711
-                          url = "https://web.archive.org/web/20200406132357/https://download.processing.org/reference.zip";
-                          sha256 = "093hc7kc9wfxqgf5dzfmfp68pbsy8x647cj0a25vgjm1swi61zbi";
-                        }
-                       } ./java/reference.zip
-
-    # suppress "Not fond of this Java VM" message box
-    substituteInPlace app/src/processing/app/platform/LinuxPlatform.java \
-      --replace 'Messages.showWarning' 'if (false) Messages.showWarning'
-
-    ( cd build
-      substituteInPlace build.xml --replace "jre-download," ""  # do not download jre1.8.0_144
-      mkdir -p linux/jre1.8.0_144                               # fake dir to avoid error
-      ant build )
+    echo "tarring jdk"
+    tar --checkpoint=10000 -czf build/linux/jdk-17.0.4-amd64.tgz ${jdk}
+    cp ${ant}/lib/ant/lib/{ant.jar,ant-launcher.jar} app/lib/
+    mkdir -p core/library
+    ln -s ${javaPackages.jogl_2_3_2}/share/java/* core/library/
+    cp ${vaqua} app/lib/VAqua9.jar
+    cp ${flatlaf} app/lib/flatlaf.jar
+    unzip -qo ${jna} -d app/lib/
+    mv app/lib/{jna-5.10.0/dist/jna.jar,}
+    mv app/lib/{jna-5.10.0/dist/jna-platform.jar,}
+    cp -r ${batik}/* java/libraries/svg/library/
+    cp java/libraries/svg/library/lib/batik-all-1.15.jar java/libraries/svg/library/batik.jar
+    echo "tarring ffmpeg"
+    tar --checkpoint=10000 -czf build/shared/tools/MovieMaker/ffmpeg-5.0.1.gz ${ffmpeg}
+    cd build
+    ant build
+    cd ..
   '';
 
   installPhase = ''
     mkdir $out
-    cp -dpR build/linux/work $out/${pname}
-
+    cp -dpr build/linux/work $out/${pname}
     rmdir $out/${pname}/java
     ln -s ${jdk} $out/${pname}/java
```

Comment:
Why $out/$pname ? Can we use something more FHS like?

Reply:
Changed to `$out/share/$pname`, this is where the Arch package puts it.